### PR TITLE
Start script + document single run

### PR DESCRIPTION
**Pull request by nikku**
This adds and documents the ability to execute our development setup in a `SINGLE_START` mode, either starting a single modeler, viewer or navigated viewer with a single CLI command.

This closes https://github.com/bpmn-io/bpmn-js/issues/1396.

### CODE REVIEW

**Comment by nikku** [comment]
As [mentioned earlier](https://github.com/bpmn-io/bpmn-js/pull/1394#issuecomment-747418737) I've consolidated the changes related to our single run feature:

* Removed from SETUP
* Added the gist to CONTRIBUTING => this is the page, that is being linked to by GitHub and should be our primary entry point for new contributors.
* Added the gist to README => Cool for some hackers to directly see I guess?

In terms of general improvement I sweep through our CONTRIBUTING guide, resulting in a few minor improvements (https://github.com/bpmn-io/bpmn-js/pull/1400/commits/bfcb6b62bf972b7cb76371dc0c734b716a50376d, https://github.com/bpmn-io/bpmn-js/pull/1400/commits/8d651ae96f4c624f920984c12a70159434dc818d).

File: package.json [hunk 1]
```diff
@@ -56,6 +58,7 @@
     "karma": "^5.2.3",
     "karma-chrome-launcher": "^3.1.0",
     "karma-coverage": "^2.0.3",
+    "karma-debug-launcher": "0.0.2",
```
Comment (MaxTru) [inline package.json]: Nice - I like how you identified this need and "just" published a library for it. 
Any reason this needs to be pinned to `0.0.2`?

File: test/config/karma.unit.js [hunk 2]
```diff
@@ -90,10 +92,15 @@ module.exports = function(karma) {
...
 
   if (collectTranslations) {
-    config.plugins = [].concat(config.plugins || [], 'karma-*', require('./translation-reporter'));
+    config.plugins = [].concat(config.plugins || ['karma-*'], require('./translation-reporter'));
     config.reporters = [].concat(config.reporters || [], 'translation-reporter');
     config.envPreprocessor = [].concat(config.envPreprocessor || [], 'COLLECT_TRANSLATIONS');
   }
 
+  if (singleStart) {
+    config.browsers = [].concat(config.browsers, [ 'Debug' ]);
```
Comment (MaxTru) [inline test/config/karma.unit.js:101]: (minor) sometimes in this file you use `.concat([ 'something' ]);` in different places we use `.concat('something');`.
Obviosuly yielding the same results, yet while we touch it, we might refactor it to be consistent. (Would help me at least when reading the code)

File: test/config/karma.unit.js [hunk 3]
```diff
@@ -90,10 +92,15 @@ module.exports = function(karma) {
   };
 
   if (collectTranslations) {
-    config.plugins = [].concat(config.plugins || [], 'karma-*', require('./translation-reporter'));
+    config.plugins = [].concat(config.plugins || ['karma-*'], require('./translation-reporter'));
```
Comment (MaxTru) [inline test/config/karma.unit.js:95]: When having `config.plugins` defined, `karma-*` would not be set, ergo the test-suite not run. Or am I missing something?

File: package.json [hunk 4]
```diff
@@ -5,6 +5,8 @@
   "scripts": {
     "all": "run-s lint test distro test:distro",
     "lint": "eslint .",
+    "start": "SINGLE_START=modeler npm run dev",
+    "start:viewer": "SINGLE_START=viewer npm run dev",
```
Comment (MaxTru) [inline package.json:9]: Any reason not to put the navigated-viewer?

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -44,15 +44,23 @@ git clone git@github.com:bpmn-io/diagram-js.git
...
+
+For details consult our in depth [setup guide](https://github.com/bpmn-io/bpmn-js/blob/master/docs/project/SETUP.md).
+
 
-// Run the test suite
-npm test
+### Build and Run the Project
 
-// Running the test suite with every file change
+```sh
```
Comment (MaxTru) [inline .github/CONTRIBUTING.md:58]: (Minor) I would put these three statements in three different blocks and add on top "You have three options to build and run the project". IMO would avoid any confusion to run these three statements one after the other

File: package.json (hunk 1 above)
Comment (MaxTru) [reply package.json]: Nevermind, I saw the later commit...

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -1,37 +1,41 @@
 # How to Contribute
 
-Great to see you! Help us out by [filing bugs or feature requests](#working-with-issues), assisting others in our [forums](https://forum.bpmn.io/) or by [contributing improvements](#contributing-improvements).
+Great to see you! Help us out by [filing bugs or feature requests](#work-with-issues), assisting others in our [forums](https://forum.bpmn.io/) or [contribute improvements](#contribute-improvements).
```
Comment (MaxTru) [inline .github/CONTRIBUTING.md:3]: Consistent formulation: `Help us out by filling bugs, assisting others, or contributing improvements`

**Review by MaxTru — CHANGES_REQUESTED** [review]
Really awesome addition. Accumlating over all bpmn-js contributors, this will save significant time and efforts.

I just have a few comments, nothing minor. In general this is good IMO.

File: test/config/karma.unit.js (hunk 3 above)
Comment (nikku) [reply test/config/karma.unit.js:95]: If defined, we need to include `karma-*`. This is the way karma works.

So if plug-ins are previously defined, they have to include `karma-*` already or nothing would work. If that makes sense :thinking:.

Comment (MaxTru) [reply test/config/karma.unit.js:95]: Got it

**Comment by nikku** [comment]
Thanks for your review hints. All incorporated :tada:

**Comment by philippfromme** [comment]
Nice!

**Pull request merged**